src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -158,22 +158,27 @@ protected string GetTestFileName(int? index = null, [CallerMemberName] string me
 
         protected static bool CreateSymLink(string targetPath, string linkPath, bool isDirectory)
         {
+#if NETFRAMEWORK
+            Process symLinkProcess = new Process();
+            symLinkProcess.StartInfo = GetWindowsStartInfo(targetPath, linkPath, isDirectory);
+#else
```

Comment:
Why this wasn't needed before?

Reply:
NETFRAMEWORK only works on Windows. So the `OperatingSystem` checks below are not available there.

Reply:
In that case, could we ifdef only the condition that doesn't work in NETFX? i.e:
```cs
#if NETFRAMEWORK
    if (false)
#else 
    if (OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsMacCatalyst()) // OSes that don't support Process.Start()
#endif

Reply:
I'm not sure I understand the suggestion. I don't see the benefit of adding an `if (false)` .

Reply:
>  I don't see the benefit of adding an if (false)

Less ifdef'd code

Reply:
I found a way to simplify it to make it more readable.